src/status-codes.ts

```diff
@@ -0,0 +1,69 @@
+import * as assert from 'assert';
+
+export function resolvedStatusCodes(codes: string[]): Map<string, number[]> {
+  const seen = new Set();
+  codes.forEach(code => {
+    assert(!seen.has(code), `Duplicate status code ${code}`);
+    assert(
+      /^\dXX$/.test(code) || /^default$/.test(code) || /^\d\d\d$/.test(code),
+      `Malformed http status code ${code}`
+    );
+    seen.add(code);
+  });
+  const record = new Map();
+  const matchers: RegExp[] = codes.sort().map(c => {
+    if (c === 'default') {
+      return /.*/;
+    }
+    return new RegExp(c.replace(/XX$/, '..'));
+  });
+  for (const code of statusCodes) {
```

Comment:
how would this behave if we have two matches? something like:
```yml
default:
  description: rest
404:
  description: not found
4XX:
  description: some error
```

all the `4XX` codes would also match `default` and might end up in either depending on the order in which they were declared in the file I guess 😣

Reply:
https://github.com/smartlyio/oats/pull/213/files#diff-4919a218fd041747e2f69f1ec399a576035ec3018691faa95eefcc02a5f264c0R14 sorts the status codes to first have numbers and then letters so 400 gets sorted before 4XX which get sorted before default.

thats also the reason for the old school loop here to allow breaking from the loop when we get the first match

Reply:
oh nice! didn't notice the `sort()` there 👍